lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -127,6 +131,14 @@ spec = do
                             \the expected 3 bytes"
                         e `shouldContain` "getMessage"
 
+        it "Regression #1025" $ do
+            let bytes = BL.fromStrict $ unsafeFromHex
+                    "005200000002d5d800000000ffffffff000000003a2954d0068a79b7fa\
+                    \44d9b75d81a78e0c527b3060e923b28d6c981e9024ca28000000000000\
+                    \0000000000000000000000000000000000000000000000000000"
+            res <- try' (runGet getBlockHeader bytes)
+            force res `shouldSatisfy` isRight
```

Comment:
Why is it necessary to force the value here?

Surely, evaluating `isRight x` will demand that `x` is evaluated to the point that we can pattern match on `Right _`, in which case the parsing must have succeeded?

Reply:
The problem isn't in the parsing however, we do parse the right number of bytes. The problem is in the conversion from `Word32` to `Word16` (fields in the `BlockHeader` aren't strict), so evaluating `BlockHeader` to WHNF doesn't force the evaluation of `slot :: SlotNo` to WHNF.

Reply:
> The problem isn't in the parsing however, we do parse the right number of bytes. The problem is in the conversion from `Word32` to `Word16` (fields in the `BlockHeader` aren't strict), so evaluating `BlockHeader` to WHNF doesn't force the evaluation of `slot :: SlotNo` to WHNF.

Fair enough. This explanation makes sense. :+1:

Reply:
Though perhaps an alternative would be to explicitly state the value that we expect, and use ``` `shouldBe` expectedValue``` instead of ``` `shouldSatisfy` isRight```.